### PR TITLE
chore: add github workflow for building apk

### PR DESCRIPTION
**Pull request by achou11**
Inspired by and built on top of #59 (because I couldn't figure out how to push updates to the PR...)

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -26,12 +25,12 @@ echo "◆ Generate FFI bindings"
 echo
 
 # See: https://cjycode.com/flutter_rust_bridge/integrate/deps.html#system-dependencies
-CPATH="$(clang -v 2>&1 | grep "Selected GCC installation" | rev | cut -d' ' -f1 | rev)/include" \
-        flutter_rust_bridge_codegen \
-                            --inline-rust \
-                            --skip-add-mod-to-lib \
-                            --rust-input packages/p2panda/native/src/api.rs \
-                            --dart-output packages/p2panda/lib/src/bridge_generated.dart
```
Comment (adzialocha) [inline scripts/build.sh:34]: I've added this to make the build work on my machine but that was with an older version of the flutter_rust_bridge, let me try if it builds without it now as well 🫣

Comment (adzialocha) [reply scripts/build.sh:34]: This is the link to the documentation (which got updated and is currently broken in the script):

https://cjycode.com/flutter_rust_bridge/v1/integrate/deps.html#system-dependencies

It says it's for Non-Debian system, so maybe it's only for me as an Arch Linux user?

I'll investigate!

**Comment by achou11** [comment]
Next steps I would like to try before merging:

- break out the rust deps part of build.sh into a separate script. these can be handled by the rust action we use so ideally they're only used when running things locally. reasoning for this is to see if caching can be taken advantage of here.

File: scripts/build.sh (hunk 1 above)
Comment (adzialocha) [reply scripts/build.sh]: Looks like it's still required for me, I've added now a check which will set the `CPATH` when using a non-debian based Linux

**Pull request merged**